agent/agent_endpoint_test.go

```diff
@@ -357,17 +375,20 @@ func TestAgent_Services_ACLFilter(t *testing.T) {
 	}
 
 	t.Run("no token", func(t *testing.T) {
-		require := require.New(t)
-
-		req := httptest.NewRequest("GET", "/v1/agent/services", nil)
-		rsp := httptest.NewRecorder()
-
-		obj, err := a.srv.AgentServices(rsp, req)
-		require.NoError(err)
+		req, _ := http.NewRequest("GET", "/v1/agent/services", nil)
+		resp := httptest.NewRecorder()
+		a.srv.h.ServeHTTP(resp, req)
+		dec := json.NewDecoder(resp.Body)
+		var val map[string]*api.AgentService
+		err := dec.Decode(&val)
+		if err != nil {
+			t.Fatalf("Err: %v", err)
+		}
 
-		val := obj.(map[string]*api.AgentService)
-		require.Empty(val)
-		require.Empty(rsp.Header().Get("X-Consul-Results-Filtered-By-ACLs"))
+		if len(val) != 0 {
+			t.Fatalf("bad: %v", val)
+		}
+		require.Empty(t, resp.Header().Get("X-Consul-Results-Filtered-By-ACLs"))
```

Comment:
Try pushing a commit reverting the changes for `TestAgent_Services_ACLFilter` (I believe these tests were added from the rebase) and see if the tests pass

Reply:
It looks like these following tests were added in the rebase:
```
TestAgentConnectCALeafCert
TestAgent_Members
TestAgent_Checks
TestAgent_Services
```
Could you try restoring them to `main` first?

Reply:
Actually, ignore that! I think I see the issue. Will comment on the PR